### ISSUE TITLE
Always prompt to save when an image is dragged in (never save to central image path)

### ISSUE THREAD

**Issue by fgimian** [marktext/marktext#2561]
### Describe your feature request

Continuing on from the discussion in #549

Currently Mark Text stores images in a central image path if a document is unsaved without warning.

The editor should really prompt you to save a file if unsaved and an image is dropped rather than defaulting to a central directory. It's so easy for images to get lost with the current system imho and it's an easy thing to miss.

At a minimum, this should be enforced when a relative assets folder is used.

### What problem does this feature solve? [optional]

This ensures that images never get painstakingly lost and always reside close to their respective document.

### Additional context [optional]

N/A

**Comment by FernandoBasso** [comment]
I agree. One thing that would be nice is to be able to specify something like `./assets` and have it store pasted image in the `assets` directory the current document folder.

```
 ~/src/haskell $ tree -CF .
.
├── assets/
│   └── 2021-07-25-08-08-53-image.png
├── fold.md
├── hs-emacs.md
├── hs-function-composition.md
├── hs-getting-started.md
├── hs-list-comprehensions.md
├── hs-lists.md
├── hs-point-free-functions.md
├── hs-precedence.md
└── lazy-eval.md
```
So, if I a `.md` file is inside the directory `haskell`, I would have a directory path of `haskell/assets`. So, relative would be relative to the current opened file, not relative to the root folder I opened Mark Text in.

![image](https://user-images.githubusercontent.com/6560620/126897035-2c32d9d3-2a29-48da-9abf-b8e3fe554a59.png)
